### PR TITLE
Display a loading spinner while waiting for treasury export! (v2)

### PR DESCRIPTION
**Pull request by ajhyndman**
I rewrote this on top of the latest UI changes, and in a way that hopefully supports all download buttons!

Screen Shot 2022-05-23 at 7 17 35 pm
Screen Shot 2022-05-23 at 7 17 28 pm
Screen Shot 2022-05-23 at 7 18 17 pm
Screen Shot 2022-05-23 at 7 17 04 pm

### CODE REVIEW

File: src/components/DownloadButton.vue [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    customClass: String,
+    classes: Object,
+    disabled: Boolean
+  },
+  mounted () {
+    window.addEventListener('focus', this.clearLoadingState)
+  },
+  destroyed () {
+    window.addEventListener('focus', this.clearLoadingState)
+  },
```
Comment (igor47) [inline src/components/DownloadButton.vue:34]: interesting. i wonder if adding a comment in the code, explaining the rationale in the PR re: detecting complete downloads, might be useful going forward?

**Review by igor47 — APPROVED** [review]
lgtm, left some minor nits

File: src/components/DownloadButton.vue [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        disabled: this.disabled || this.isLoading
+      }
+    }
+  },
+  methods: {
+    clearLoadingState () {
+      this.isLoading = false
+    },
+    setLoadingState () {
+      console.log('setLoadingState()')
```
[the remark this answers is not in the document]
Comment (ajhyndman) [reply src/components/DownloadButton.vue:53]: Whoops, thanks!

File: src/components/DownloadButton.vue [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    @click="setLoadingState"
+  >
+    <span
+      v-if="isLoading"
+      class="spinner-border spinner-border-sm"
+      role="status"
+      aria-hidden="true"
+    ></span>
+    <span v-if="isLoading"> Loading...</span>
+    <span v-else> <slot /></span>
```
[the remark this answers is not in the document]
Comment (ajhyndman) [reply src/components/DownloadButton.vue:16]: This is actually intentional. Vue seems to elide any whitespace at the beginning and end of lines, but we need to ensure that there's a space between the actual button text and the loading spinner.

I agree the intent isn't clear here though.  Open to suggestions!

File: src/components/DownloadButton.vue (hunk 1 above)
Comment (ajhyndman) [reply src/components/DownloadButton.vue:34]: Oh yeah, that's a good idea.

**Pull request merged**